src/validationinterface.h

```diff
@@ -56,10 +84,55 @@ class CValidationInterface {
     friend void ::UnregisterAllValidationInterfaces();
 };
 
+// aggregate_non_null_values is a combiner which places any non-nullptr values
+// returned from slots into a container.
+template<typename Container>
+struct aggregate_non_null_values
+{
+    typedef Container result_type;
+
+    template<typename InputIterator>
+    Container operator()(InputIterator first, InputIterator last) const
+    {
+        Container values;
+
+        while (first != last) {
+            auto ptr = *first;
+            if (ptr != nullptr) {
+                values.push_back(ptr);
+            }
+            ++first;
+        }
+        return values;
+    }
+};
+
 struct CMainSignals {
     /** Notifies listeners of updated block chain tip */
     boost::signals2::signal<void (const CBlockIndex *)> UpdatedBlockTip;
-    /** Notifies listeners of updated transaction data (transaction, and optionally the block it is found in. */
+    /**
+     * Requests a pointer to the listener's batch scanner for shielded outputs,
+     * if it has one.
+     *
+     * The listener is responsible for managing the memory of the batch scanner.
+     * In practice each listener will have a single persistent batch scanner.
+     *
+     * This is an opportunity for the listener to perform any updating of the
+     * batch scanner's internal state (such as updating its set of incoming
+     * viewing keys).
```

Comment:
Document when `GetBatchScanner` is called (I think it is whenever we move between blocks).